// vite/dist/node/chunks/dep-BK3b2jBa.js
import { useMemo, useState } from "react";
import { motion, AnimatePresence } from "framer-motion";
import { ArrowLeft, ArrowUpRight, CornerDownRight } from "lucide-react";

/**
 * PORTFOLIO — TYPO “PROJECTS” LIST (LIKE YOUR REF) BUT:
 * ✅ NO “preview rail”
 * ✅ NO empty space: right side becomes a full-height “living poster”
 * ✅ Unique + attractive: moving grain, spotlight, animated title marquee, year spine
 * ✅ Flow:
 *   Root: 2 studios → Art Heir | Blue Focus
 *   Art Heir: show all projects list
 *   Blue Focus: show 3 categories → then list
 * ✅ Back button built-in
 *
 * Replace the sample data with your real images/titles.
 */

const BRAND = {
  ART_HEIR: "Art Heir Events & Exhibitions",
  BLUE_FOCUS: "Blue Focus",
};

const BLUE_CATS = ["Events", "Exhibitions", "Kiosks"];

/* -------------------- SAMPLE DATA (REPLACE) -------------------- */
const data = {
  [BRAND.ART_HEIR]: [
    {
      id: "ah-1",
      year: 2025,
      title: "Singapore Odyssea: A Journey Through Time",
      subtitle: "National Museum of Singapore",
      location: "Singapore",
      date: "2025",
      image:
        "https://images.unsplash.com/photo-1526481280695-3c687fd643ed?auto=format&fit=crop&w=2000&q=80",
    },
    {
      id: "ah-2",
      year: 2025,
      title: "Pulsations",
      subtitle: "Pavillon France — Expo 2025",
      location: "Osaka, Japan",
      date: "2025",
      image:
        "https://images.unsplash.com/photo-1526948128573-703ee1aeb6fa?auto=format&fit=crop&w=2000&q=80",
    },
    {
      id: "ah-3",
      year: 2025,
      title: "Nanualuk — Northern Expedition",
      subtitle: "Montreal Science Centre",
      location: "Montreal, Canada",
      date: "2025",
      image:
        "https://images.unsplash.com/photo-1450849608880-6f787542c88a?auto=format&fit=crop&w=2000&q=80",
    },
    {
      id: "ah-4",
      year: 2024,
      title: "Light: Visionary Perspectives",
      subtitle: "Aga Khan Museum",
      location: "Toronto, Canada",
      date: "2024",
      image:
        "https://images.unsplash.com/photo-1520697222865-7f97a36b8f08?auto=format&fit=crop&w=2000&q=80",
    },
  ],
  [BRAND.BLUE_FOCUS]: {
    Events: [
      {
        id: "bf-e-1",
        year: 2025,
        title: "Signature Corporate Launch",
        subtitle: "Dubai World Trade Centre",
        location: "Dubai, UAE",
        date: "2025",
        image:
          "https://images.unsplash.com/photo-1521737604893-d14cc237f11d?auto=format&fit=crop&w=2000&q=80",
      },
      {
        id: "bf-e-2",
        year: 2024,
        title: "Luxury Gala Night",
        subtitle: "Atlantis The Royal",
        location: "Dubai, UAE",
        date: "2024",
        image:
          "https://images.unsplash.com/photo-1527529482837-4698179dc6ce?auto=format&fit=crop&w=2000&q=80",
      },
    ],
    Exhibitions: [
      {
        id: "bf-x-1",
        year: 2025,
        title: "Expo Stand Experience",
        subtitle: "Custom Build + AV + Branding",
        location: "Abu Dhabi, UAE",
        date: "2025",
        image:
          "https://images.unsplash.com/photo-1540575467063-178a50c2df87?auto=format&fit=crop&w=2000&q=80",
      },
      {
        id: "bf-x-2",
        year: 2024,
        title: "Tech Showcase Pavilion",
        subtitle: "Immersive Wall + Lighting",
        location: "Dubai, UAE",
        date: "2024",
        image:
          "https://images.unsplash.com/photo-1515169067868-5387ec356754?auto=format&fit=crop&w=2000&q=80",
      },
    ],
    Kiosks: [
      {
        id: "bf-k-1",
        year: 2025,
        title: "Retail Kiosk Concept Series",
        subtitle: "Modular + Premium Finishes",
        location: "Dubai, UAE",
        date: "2025",
        image:
          "https://images.unsplash.com/photo-1555529669-e69e7aa0ba9a?auto=format&fit=crop&w=2000&q=80",
      },
      {
        id: "bf-k-2",
        year: 2024,
        title: "Mall Activation Kiosk",
        subtitle: "Brand Pop-up + Sampling",
        location: "Sharjah, UAE",
        date: "2024",
        image:
          "https://images.unsplash.com/photo-1520975693411-b2ce41a1a6a8?auto=format&fit=crop&w=2000&q=80",
      },
    ],
  },
};
/* --------------------------------------------------------------- */

const ease = [0.2, 0.8, 0.2, 1];

const fade = {
  hidden: { opacity: 0, y: 12 },
  show: { opacity: 1, y: 0, transition: { duration: 0.5, ease } },
  exit: { opacity: 0, y: 10, transition: { duration: 0.25, ease } },
};

export default function PortfolioProjectsUltra({
  onNavigate = () => {},
  dataset = data,
}) {
  /**
   * view:
   *  "root" | "artheir" | "blueCats" | "blueList"
   */
  const [view, setView] = useState("root");
  const [blueCat, setBlueCat] = useState("Events");
  const [activeId, setActiveId] = useState(null);

  const artHeir = dataset[BRAND.ART_HEIR] || [];
  const blue = dataset[BRAND.BLUE_FOCUS] || {};
  const blueList = blue?.[blueCat] || [];

  const list = useMemo(() => {
    if (view === "artheir") return artHeir;
    if (view === "blueList") return blueList;
    return [];
  }, [view, artHeir, blueList]);

  // pick a “poster” project (active hover/selection, else first)
  const poster = useMemo(() => {
    if (!list?.length) return null;
    return list.find((p) => p.id === activeId) || list[0];
  }, [list, activeId]);

  const headerLine =
    view === "root"
      ? "Select a studio to explore"
      : view === "artheir"
      ? BRAND.ART_HEIR
      : view === "blueCats"
      ? BRAND.BLUE_FOCUS
      : `${BRAND.BLUE_FOCUS} — ${blueCat}`;

  const goBack = () => {
    if (view === "artheir") return setView("root");
    if (view === "blueCats") return setView("root");
    if (view === "blueList") return setView("blueCats");
    setView("root");
  };

  // make sure an active poster exists when entering list views
  const enterList = (nextView) => {
    setView(nextView);
    const nextList = nextView === "artheir" ? artHeir : blueList;
    setActiveId(nextList?.[0]?.id ?? null);
  };

  return (
    <section className="relative bg-white overflow-hidden">
      {/* Ultra background (bold but clean) */}
      <div className="absolute inset-0 pointer-events-none">
        {/* spotlights */}
        <div className="absolute -top-52 left-1/2 -translate-x-1/2 w-[1200px] h-[1200px] bg-[#206ca6]/12 rounded-full blur-3xl" />
        <div className="absolute -bottom-80 right-[-420px] w-[1100px] h-[1100px] bg-[#b1b4ad]/18 rounded-full blur-3xl" />
        {/* grain */}
        <div className="absolute inset-0 opacity-[0.06] bg-[radial-gradient(#0f172a_1px,transparent_1px)] [background-size:26px_26px]" />
        {/* watermark */}
        <div className="absolute top-8 left-1/2 -translate-x-1/2 text-[84px] sm:text-[150px] lg:text-[240px] font-extrabold tracking-tight text-gray-900/5 select-none whitespace-nowrap">
          PROJECTS
        </div>
      </div>

      <div className="relative z-10 max-w-7xl mx-auto px-6 lg:px-8 pt-24 pb-28">
        {/* Top heading row */}
        <div className="flex items-start justify-between gap-6">
          <div>
            <motion.h2
              initial="hidden"
              animate="show"
              variants={fade}
              className="text-5xl sm:text-6xl font-extrabold tracking-tight text-gray-900"
            >
              Projects
            </motion.h2>
            <motion.p
              initial="hidden"
              animate="show"
              variants={fade}
              transition={{ delay: 0.06 }}
              className="mt-3 text-gray-600 text-base sm:text-lg"
            >
              {headerLine}
            </motion.p>
          </div>

          <div className="flex items-center gap-3">
            {view !== "root" && (
              <button
                onClick={goBack}
                className="group inline-flex items-center gap-2 rounded-full border border-gray-200 bg-white/70 backdrop-blur px-4 py-2 text-sm font-semibold text-gray-800 hover:border-[#206ca6]/40 hover:text-[#206ca6] transition"
              >
                <ArrowLeft className="w-4 h-4 group-hover:-translate-x-0.5 transition-transform" />
                Back
              </button>
            )}
            <button
              onClick={() => onNavigate("contact")}
              className="hidden sm:inline-flex items-center gap-2 rounded-full bg-gray-900 text-white px-4 py-2 text-sm font-semibold hover:bg-gray-800 transition"
            >
              Start a Project
              <ArrowUpRight className="w-4 h-4" />
            </button>
          </div>
        </div>

        {/* Divider */}
        <div className="mt-10 h-px w-full bg-gradient-to-r from-transparent via-gray-200 to-transparent" />

        {/* FULL / NO SPACE LAYOUT */}
        <div className="mt-12 grid lg:grid-cols-12 gap-10 items-stretch">
          {/* LEFT: controls + list (takes full height) */}
          <div className="lg:col-span-7 flex flex-col min-h-[620px]">
            <AnimatePresence mode="wait">
              {view === "root" && (
                <motion.div
                  key="root"
                  variants={fade}
                  initial="hidden"
                  animate="show"
                  exit="exit"
                  className="h-full"
                >
                  <StudioTiles
                    leftTitle={BRAND.ART_HEIR}
                    rightTitle={BRAND.BLUE_FOCUS}
                    onLeft={() => enterList("artheir")}
                    onRight={() => setView("blueCats")}
                  />
                </motion.div>
              )}

              {view === "blueCats" && (
                <motion.div
                  key="blueCats"
                  variants={fade}
                  initial="hidden"
                  animate="show"
                  exit="exit"
                  className="h-full flex flex-col"
                >
                  {/* Big category selector (unique: split panels) */}
                  <div className="grid sm:grid-cols-3 gap-5">
                    {BLUE_CATS.map((c) => {
                      const active = c === blueCat;
                      return (
                        <button
                          key={c}
                          onClick={() => setBlueCat(c)}
                          className={`group relative text-left rounded-[28px] border bg-white/70 backdrop-blur p-7 transition overflow-hidden
                            ${
                              active
                                ? "border-[#206ca6]/35 shadow-[0_40px_110px_-70px_rgba(32,108,166,0.6)]"
                                : "border-gray-200 hover:border-[#206ca6]/25 hover:bg-white"
                            }`}
                        >
                          {/* top accent */}
                          <div
                            className={`absolute inset-x-0 top-0 h-1 ${
                              active ? "bg-[#206ca6]" : "bg-gray-200"
                            }`}
                          />
                          <div className="text-xs font-semibold tracking-[0.34em] uppercase text-gray-500">
                            Category
                          </div>
                          <div
                            className={`mt-2 text-2xl font-extrabold tracking-tight ${
                              active ? "text-[#206ca6]" : "text-gray-900"
                            }`}
                          >
                            {c}
                          </div>
                          <div className="mt-2 text-gray-600 text-sm leading-relaxed">
                            Explore {c} projects in an editorial list.
                          </div>
                          <div className="mt-6 inline-flex items-center gap-2 text-sm font-extrabold text-gray-900">
                            Select
                            <CornerDownRight className="w-4 h-4 group-hover:translate-x-0.5 group-hover:translate-y-0.5 transition-transform" />
                          </div>
                        </button>
                      );
                    })}
                  </div>

                  <button
                    onClick={() => {
                      setView("blueList");
                      setActiveId(blueList?.[0]?.id ?? null);
                    }}
                    className="mt-6 group w-full text-left rounded-[28px] border border-gray-200 bg-gray-900 text-white p-7 hover:bg-gray-800 transition"
                  >
                    <div className="flex items-start justify-between gap-4">
                      <div>
                        <div className="text-xs font-semibold tracking-[0.34em] uppercase text-white/65">
                          {BRAND.BLUE_FOCUS}
                        </div>
                        <div className="mt-2 text-2xl sm:text-3xl font-extrabold">
                          Open {blueCat} List
                        </div>
                        <div className="mt-2 text-white/70">
                          Clean typography, like your reference — but more premium.
                        </div>
                      </div>
                      <ArrowUpRight className="w-6 h-6 text-white/70 group-hover:text-white group-hover:translate-x-0.5 group-hover:-translate-y-0.5 transition-all" />
                    </div>
                  </button>
                </motion.div>
              )}

              {(view === "artheir" || view === "blueList") && (
                <motion.div
                  key={view}
                  variants={fade}
                  initial="hidden"
                  animate="show"
                  exit="exit"
                  className="h-full flex flex-col"
                >
                  <ProjectsTypoList
                    projects={list}
                    activeId={activeId}
                    setActiveId={setActiveId}
                  />
                </motion.div>
              )}
            </AnimatePresence>
          </div>

          {/* RIGHT: FULL-HEIGHT “LIVING POSTER” (NO SPACE LEFT) */}
          <div className="lg:col-span-5">
            <LivingPoster
              view={view}
              poster={poster}
              fallbackTitle={view === "root" ? "Choose a studio" : headerLine}
              onPrimary={() => onNavigate("contact")}
              onSecondary={() => onNavigate("services")}
            />
          </div>
        </div>
      </div>
    </section>
  );
}

/* -------------------- UNIQUE RIGHT SIDE: LIVING POSTER -------------------- */

function LivingPoster({ view, poster, fallbackTitle, onPrimary, onSecondary }) {
  const hasPoster = !!poster;

  return (
    <div className="h-full min-h-[620px] lg:min-h-[680px] rounded-[34px] border border-gray-200 bg-white/70 backdrop-blur overflow-hidden shadow-[0_60px_140px_-90px_rgba(15,23,42,0.55)]">
      <div className="relative h-full">
        {/* image */}
        <AnimatePresence mode="wait">
          <motion.div
            key={hasPoster ? poster.id : "empty"}
            initial={{ opacity: 0, scale: 1.03 }}
            animate={{ opacity: 1, scale: 1 }}
            exit={{ opacity: 0, scale: 1.02 }}
            transition={{ duration: 0.55, ease }}
            className="absolute inset-0"
          >
            <img
              src={
                hasPoster
                  ? poster.image
                  : "https://images.unsplash.com/photo-1540575467063-178a50c2df87?auto=format&fit=crop&w=2000&q=80"
              }
              alt={hasPoster ? poster.title : "poster"}
              className="h-full w-full object-cover"
              loading="lazy"
            />
            {/* cinematic overlays */}
            <div className="absolute inset-0 bg-gradient-to-t from-black/68 via-black/18 to-transparent" />
            <div className="absolute inset-0 bg-[radial-gradient(circle_at_30%_15%,rgba(255,255,255,0.14),transparent_55%)]" />
            <div className="absolute inset-0 opacity-[0.09] bg-[linear-gradient(to_bottom,rgba(255,255,255,0.18)_1px,transparent_1px)] [background-size:100%_10px]" />
          </motion.div>
        </AnimatePresence>

        {/* animated marquee text (fills space, looks premium) */}
        <div className="absolute inset-x-0 top-4 overflow-hidden pointer-events-none">
          <motion.div
            initial={{ x: 0 }}
            animate={{ x: -520 }}
            transition={{ repeat: Infinity, duration: 16, ease: "linear" }}
            className="whitespace-nowrap text-white/14 font-extrabold tracking-tight text-[44px]"
          >
            {hasPoster ? poster.title : "PORTFOLIO"} •{" "}
            {hasPoster ? poster.subtitle : "STUDIO WORK"} •{" "}
            {hasPoster ? poster.location : "EVENTS"} •{" "}
            {hasPoster ? poster.title : "PROJECTS"} •{" "}
            {hasPoster ? poster.subtitle : "EXHIBITIONS"} •{" "}
            {hasPoster ? poster.location : "KIOSKS"} •{" "}
          </motion.div>
        </div>

        {/* content */}
        <div className="relative h-full flex flex-col justify-end p-7 sm:p-8">
          {/* year chip */}
          <div className="mb-4 flex items-center gap-3">
            <span className="inline-flex items-center rounded-full bg-white/12 border border-white/18 px-3 py-1 text-xs font-extrabold tracking-[0.34em] uppercase text-white/85">
              {hasPoster ? poster.year : "—"}
            </span>
            <span className="text-white/70 text-xs font-semibold tracking-[0.32em] uppercase">
              {view === "root"
                ? "Select Studio"
                : view === "artheir"
                ? "Art Heir"
                : "Blue Focus"}
            </span>
          </div>

          <div className="text-white">
            <div className="text-2xl sm:text-3xl font-extrabold leading-tight">
              {hasPoster ? poster.title : fallbackTitle}
            </div>
            <div className="mt-2 text-white/80 text-base sm:text-lg">
              {hasPoster
                ? poster.subtitle
                : "A bold editorial layout — no cards, no clutter."}
            </div>

            <div className="mt-4 flex flex-wrap gap-2">
              <span className="inline-flex items-center rounded-full bg-white/10 border border-white/15 px-3 py-1 text-sm text-white/85">
                {hasPoster ? poster.location : "Dubai • Mumbai • Global"}
              </span>
              <span className="inline-flex items-center rounded-full bg-white/10 border border-white/15 px-3 py-1 text-sm text-white/85">
                {hasPoster ? poster.date : "High-impact production"}
              </span>
            </div>
          </div>

          <div className="mt-6 grid grid-cols-2 gap-3">
            <button
              onClick={onPrimary}
              className="rounded-2xl bg-white text-gray-900 font-extrabold py-3.5 hover:bg-gray-100 transition"
            >
              Work With Us
            </button>
            <button
              onClick={onSecondary}
              className="rounded-2xl border border-white/28 bg-white/10 text-white font-extrabold py-3.5 hover:bg-white/15 transition"
            >
              View Services
            </button>
          </div>

          {/* bottom mini “signature” */}
          <div className="mt-6 flex items-center justify-between text-xs text-white/70">
            <span className="font-semibold tracking-[0.34em] uppercase">
              Crafted for premium portfolios
            </span>
            <span className="opacity-80">Scroll / Hover to preview</span>
          </div>
        </div>
      </div>
    </div>
  );
}

/* -------------------- ROOT: STUDIO TILES -------------------- */

function StudioTiles({ leftTitle, rightTitle, onLeft, onRight }) {
  return (
    <div className="grid sm:grid-cols-2 gap-6 h-full">
      <button
        onClick={onLeft}
        className="group relative text-left rounded-[32px] border border-gray-200 bg-white/70 backdrop-blur p-9 overflow-hidden hover:border-[#206ca6]/35 hover:shadow-[0_55px_130px_-90px_rgba(32,108,166,0.75)] transition"
      >
        <div className="absolute inset-0 opacity-0 group-hover:opacity-100 transition">
          <div className="absolute -top-28 -left-28 w-[420px] h-[420px] bg-[#206ca6]/18 rounded-full blur-3xl" />
        </div>

        <div className="text-xs font-semibold tracking-[0.34em] uppercase text-gray-500">
          Studio 01
        </div>
        <div className="mt-3 text-2xl sm:text-3xl font-extrabold tracking-tight text-gray-900">
          {leftTitle}
        </div>
        <div className="mt-3 text-gray-600 text-sm leading-relaxed">
          Open a clean typography list (like the reference image) with bold poster preview.
        </div>

        <div className="mt-7 inline-flex items-center gap-2 text-sm font-extrabold text-[#206ca6]">
          Open
          <ArrowUpRight className="w-4 h-4 group-hover:translate-x-0.5 group-hover:-translate-y-0.5 transition-transform" />
        </div>
      </button>

      <button
        onClick={onRight}
        className="group relative text-left rounded-[32px] border border-gray-200 bg-gray-900 text-white p-9 overflow-hidden hover:bg-gray-800 transition"
      >
        <div className="absolute inset-0 opacity-100">
          <div className="absolute -bottom-40 -right-40 w-[520px] h-[520px] bg-white/10 rounded-full blur-3xl" />
          <div className="absolute inset-0 opacity-[0.08] bg-[radial-gradient(#ffffff_1px,transparent_1px)] [background-size:24px_24px]" />
        </div>

        <div className="relative">
          <div className="text-xs font-semibold tracking-[0.34em] uppercase text-white/65">
            Studio 02
          </div>
          <div className="mt-3 text-2xl sm:text-3xl font-extrabold tracking-tight">
            {rightTitle}
          </div>
          <div className="mt-3 text-white/70 text-sm leading-relaxed">
            Choose a section (Events / Exhibitions / Kiosks) then view the editorial list.
          </div>

          <div className="mt-7 inline-flex items-center gap-2 text-sm font-extrabold text-white">
            Select Category
            <ArrowUpRight className="w-4 h-4 group-hover:translate-x-0.5 group-hover:-translate-y-0.5 transition-transform" />
          </div>
        </div>
      </button>
    </div>
  );
}

/* -------------------- TYPO LIST (LIKE REF) + YEAR SPINE -------------------- */

function ProjectsTypoList({ projects, activeId, setActiveId }) {
  const grouped = useMemo(() => {
    const map = new Map();
    projects.forEach((p) => {
      const y = p.year || "—";
      if (!map.has(y)) map.set(y, []);
      map.get(y).push(p);
    });
    const entries = Array.from(map.entries()).sort((a, b) => Number(b[0]) - Number(a[0]));
    return entries;
  }, [projects]);

  return (
    <div className="relative flex-1">
      {/* Left vertical spine for “no empty feel” */}
      <div className="absolute left-0 top-0 bottom-0 w-px bg-gradient-to-b from-transparent via-gray-200 to-transparent" />

      <div className="pl-6">
        {grouped.map(([year, items]) => (
          <div key={year} className="py-4">
            {items.map((p, idx) => {
              const active = p.id === activeId;
              return (
                <button
                  key={p.id}
                  type="button"
                  onMouseEnter={() => setActiveId(p.id)}
                  onFocus={() => setActiveId(p.id)}
                  onClick={() => setActiveId(p.id)}
                  className="group w-full text-left"
                >
                  <div className="grid grid-cols-12 gap-6 items-start py-7">
                    {/* YEAR column */}
                    <div className="col-span-12 sm:col-span-2">
                      <div className="text-gray-900 font-extrabold text-lg sm:text-xl">
                        {idx === 0 ? year : ""}
                      </div>
                    </div>

                    {/* Title */}
                    <div className="col-span-12 sm:col-span-10">
                      <div className="flex items-start justify-between gap-4">
                        <div>
                          <div
                            className={`text-2xl sm:text-3xl font-extrabold tracking-tight transition-colors ${
                              active ? "text-[#206ca6]" : "text-gray-900 group-hover:text-[#206ca6]"
                            }`}
                          >
                            {p.title}
                          </div>
                          <div className="mt-1 text-base sm:text-lg text-gray-600">
                            {p.subtitle}
                          </div>
                        </div>

                        <ArrowUpRight
                          className={`w-6 h-6 shrink-0 transition-all ${
                            active
                              ? "text-[#206ca6] translate-x-0.5 -translate-y-0.5"
                              : "text-gray-300 group-hover:text-[#206ca6] group-hover:translate-x-0.5 group-hover:-translate-y-0.5"
                          }`}
                        />
                      </div>

                      {/* micro meta line (optional) */}
                      <div className="mt-4 text-sm text-gray-500">
                        <span className="font-semibold text-gray-700">{p.location}</span>
                        <span className="mx-2 text-gray-300">•</span>
                        <span>{p.date}</span>
                      </div>

                      {/* active underline (super clean) */}
                      <div
                        className={`mt-6 h-px w-full transition ${
                          active
                            ? "bg-gradient-to-r from-[#206ca6]/40 via-gray-200 to-transparent"
                            : "bg-gradient-to-r from-transparent via-gray-200 to-transparent"
                        }`}
                      />
                    </div>
                  </div>
                </button>
              );
            })}
          </div>
        ))}
      </div>
    </div>
  );
}
